content/blog/2016/2016-05-19-update-plugin-for-pipeline-tutorial.md

```diff
@@ -0,0 +1,189 @@
+---
+:layout: post
+:title: Refactoring a Jenkins plugin for compatibility with Pipeline jobs
+:tags:
+- core
+:author: cprice404
+---
+
+NOTE: _This is a guest post by [Chris Price](https://github.com/cprice404).
+Chris is a software engineer at [Puppet](http://puppet.com), and has been
+spending some time lately on automating performance testing using the latest
+Jenkins features._
+
+## Adventures in Jenkins plugin development: Pipeline refactoring edition
+
+In this blog post, I'm going to attempt to provide some step-by-step notes on how to refactor an existing Jenkins plugin to make it compatible with the new Jenkins Pipeline jobs.  Before we get to the fun stuff, though, a little background.
+
+### How'd I end up here?
+
+Recently, I started working on a project to automate some performance tests for my company's products.  We use the awesome [Gatling](http://gatling.io/#/) load testing tool for these tests, but we've largely been handling the testing very manually to date, due to a lack of bandwidth to get them automated in a clean, maintainable, extensible way.  We have a years-old Jenkins server where we use the [gatling jenkins plugin](https://wiki.jenkins-ci.org/display/JENKINS/Gatling+Plugin) to track the history of certain tests over time, but the setup of the Jenkins instance was very delicate and not easy to reproduce, so it had fallen into a state of disrepair.
```

Comment:
I have a penchant for run-on sentences.  :)  Happy to break up anything that needs breaking up.

I'm also fine telling my editor to insert hard line wraps at whatever line size you prefer, but made the choice not to do so in the original draft after looking at a few previous blog posts and observing that they appeared not to be doing so.  Also for the concerns about how it would affect rendering.

In any case, happy to make whatever changes are needed on that front, just let me know what the consensus is.